rust/connlib/tunnel/src/tests/sim_client.rs

```diff
@@ -413,7 +418,11 @@ impl RefClient {
 
         // If we have a resource, the first packet will initiate a connection to the gateway.
         tracing::debug!("Not connected to resource, expecting to trigger connection intent");
-        self.connected_cidr_resources.insert(rid);
+        if self.internet_resource.as_ref().map(|r| r.id) == Some(rid) {
```

Comment:
it's a bit akward because to make a method general like that we would need to pass the `DomainName` since `connected_dns_resource` takes a domain name and id tuple, I think it's easier like this.

Reply:
I have resolved it by making a less general `connect_to_cidr_or_internet_resource` is a bit weird but makes it easier to read I think